src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -1292,6 +1399,1821 @@ public unsafe record C
+        public void RecordProperties_PropertyInValueTuple_Static()
+        {
+            var corlib_cs = @"
+namespace System
+{
+    public class Object { }
+    public class ValueType
+    {
+        public static bool X { get; set; }
+    }
+    public class Attribute { }
+    public struct Void { }
+    public struct Boolean { }
+    public class Exception { }
+}
+";
+            var corlibRef = CreateEmptyCompilation(corlib_cs).EmitToImageReference();
+
+            var src = @"
+record struct C(bool X)
+{
+    bool M()
+    {
+        return X;
+    }
+}
+";
+            var comp = CreateEmptyCompilation(src, parseOptions: TestOptions.RegularPreview, references: new[] { corlibRef });
+            comp.VerifyEmitDiagnostics(
+                // (2,22): error CS8866: Record member 'System.ValueType.X' must be a readable instance property of type 'bool' to match positional parameter 'X'.
+                // record struct C(bool X)
+                Diagnostic(ErrorCode.ERR_BadRecordMemberForPositionalParameter, "X").WithArguments("System.ValueType.X", "bool", "X").WithLocation(2, 22),
+                // (2,22): warning CS8907: Parameter 'X' is unread. Did you forget to use it to initialize the property with that name?
+                // record struct C(bool X)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "X").WithArguments("X").WithLocation(2, 22)
+                );
+        }
+
+        [Fact]
+        public void RecordProperties_PropertyInValueTuple_Readonly()
+        {
+            var corlib_cs = @"
+namespace System
+{
+    public class Object { }
+    public class ValueType
+    {
+        public bool X { get; set; }
+    }
+    public class Attribute { }
+    public struct Void { }
+    public struct Boolean { }
+}
+";
+            var corlibRef = CreateEmptyCompilation(corlib_cs).EmitToImageReference();
+
+            var src = @"
+readonly record struct C(bool X)
+{
+    bool M()
+    {
+        return X;
+    }
+}
+";
+            var comp = CreateEmptyCompilation(src, parseOptions: TestOptions.RegularPreview, references: new[] { corlibRef });
+            comp.VerifyEmitDiagnostics(
+                // (2,31): warning CS8907: Parameter 'X' is unread. Did you forget to use it to initialize the property with that name?
+                // readonly record struct C(bool X);
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "X").WithArguments("X").WithLocation(2, 31)
+                );
+
+            Assert.Null(comp.GlobalNamespace.GetTypeMember("C").GetMember("X"));
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var x = tree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().Single().Expression;
+            Assert.Equal("System.Boolean System.ValueType.X { get; set; }", model.GetSymbolInfo(x!).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void StaticCtor()
```

Comment:
>StaticCtor [](start = 20, length = 10)

It is not clear what are we testing here. Could you elaborate please?

Reply:
I messed up that test...

---
In reply to: [590818547](https://github.com/dotnet/roslyn/pull/51720#discussion_r590818547) [](ancestors = 590818547)